python/rikai/types/vision.py

```diff
@@ -81,10 +88,12 @@ def from_array(
 
         assert array is not None
         img = PILImage.fromarray(array, mode=mode)
-        return cls.from_pil(img, uri)
+        return cls.from_pil(img, uri, format=format, **kwargs)
 
     @staticmethod
-    def from_pil(img: PILImage, uri: Union[str, Path]) -> Image:
+    def from_pil(
```

Comment:
giving that we will support various types from `__init__` , we might be able to consolidate this and `from_array` and etc factory methods together.

Not need to be done here tho.